src/sources/opentelemetry/http.rs

```diff
@@ -21,13 +22,21 @@ use vector_lib::tls::MaybeTlsIncomingStream;
 use vector_lib::{
     config::LogNamespace,
     event::{BatchNotifier, BatchStatus},
+    lookup::path,
     EstimatedJsonEncodedSizeOf,
 };
-use warp::{filters::BoxedFilter, reject::Rejection, reply::Response, Filter, Reply};
+use warp::{
+    filters::BoxedFilter,
+    http::{HeaderMap, HeaderValue},
+    reject::Rejection,
+    reply::Response,
+    Filter, Reply,
+};
 
 use crate::http::{KeepaliveConfig, MaxConnectionAgeLayer};
+use crate::sources::http_server::{HttpConfigParamKind, SimpleHttpConfig};
```

Comment:
Not entirely sure if it's OK to use types from the `source::http_server` module, but other sources (e.g. `heroku_logs`) also do it, so I assumed it is OK.

Reply:
We can also move `HttpConfigParamKind` into the common file.

Reply:
I decided against making this change for now, as it touches a number of unrelated files.